src/renderer/components/dock/pod-logs.store.ts

```diff
@@ -1,33 +1,38 @@
-import { autorun, observable } from "mobx";
-import { Pod, IPodContainer, podsApi } from "../../api/endpoints";
+import { autorun, computed, observable, reaction } from "mobx";
+import { Pod, IPodContainer, podsApi, IPodLogsQuery } from "../../api/endpoints";
 import { autobind, interval } from "../../utils";
 import { DockTabStore } from "./dock-tab.store";
 import { dockStore, IDockTab, TabKind } from "./dock.store";
 import { t } from "@lingui/macro";
 import { _i18n } from "../../i18n";
+import { Notifications } from "../notifications";
+import { isDevelopment } from "../../../common/vars";
 
 export interface IPodLogsData {
   pod: Pod;
   selectedContainer: IPodContainer
   containers: IPodContainer[]
   initContainers: IPodContainer[]
   showTimestamps: boolean
-  tailLines: number
   previous: boolean
 }
 
 type TabId = string;
+type PodLogs = string;
 
-interface PodLogs {
-  oldLogs?: string
-  newLogs?: string
-}
+// Number for log lines to load
+export const logRange = isDevelopment ? 100 : 1000;
 
 @autobind()
 export class PodLogsStore extends DockTabStore<IPodLogsData> {
-  private refresher = interval(10, () => this.load(dockStore.selectedTabId));
+  private refresher = interval(10, () => {
+    const id = dockStore.selectedTabId
+    if (!this.logs.get(id)) return
+    this.loadMore(id)
+  });
 
   @observable logs = observable.map<TabId, PodLogs>();
```

Comment:
Why are logs not the following:

```typescript
type PodLogLine = string;
...
@observable logs = observable.map<TabId, PodLogLine[]>();
```

That way the new/old boundary could easily just be an index into that array. It would also mean that the call to `lines()` doesn't have to split the entire text every time.

Is there something that I am missing here as to why storing the entire logs as one string is better?

Reply:
Good catch! I was thinking to make those transition to log array in next PRs, but nothing prevents you from doing it now.